### ISSUE TITLE
ipptool: specifying "-P" with "-t" causes extraneous output in CUPS test output content

### ISSUE THREAD

**Issue by wifiprintguy** [istopwg/ippsample#46]
If ipptool is called with the "-P" argument (to generate a test results .plist file) in addition to the "-t" argument (to have CUPS test report output used), there will be extraneous information written to stdout so that the test report output becomes polluted and difficult to read.

Example without the "-P" argument:

	Serenity: ippsample [714]$ ./tools/ipptool -vt 'ipp://Serenity.local:8991/ipp/print/_IPP-Everywhere-Basic_' test-i9-trimmed-2.test
	"test-i9-trimmed-2.test":
		Get-Printer-Attributes:
			attributes-charset (charset) = utf-8
			attributes-natural-language (naturalLanguage) = en
			printer-uri (uri) = ipp://Serenity.local:8991/ipp/print/_IPP-Everywhere-Basic_
			requesting-user-name (nameWithoutLanguage) = smitty
			requested-attributes (keyword) = printer-icons
		I-9. Get-Printer-Attributes Operation to Serenity.local              [FAIL]
			RECEIVED: 161 bytes in response
			status-code = successful-ok (successful-ok)
			attributes-charset (charset) = utf-8
			attributes-natural-language (naturalLanguage) = en
			printer-icons (uri) = https://serenity.local.:8991/ipp/print/_IPP-Everywhere-Basic_/icon.png
			EXPECTED: printer-icons WITH-ALL-VALUES "Serenity.local"
			GOT: printer-icons="https://serenity.local.:8991/ipp/print/_IPP-Everywhere-Basic_/icon.png"

Example with "-P" added:

	Serenity: ippsample [715]$ ./tools/ipptool -vt 'ipp://Serenity.local:8991/ipp/print/_IPP-Everywhere-Basic_' -P '_IPP-Everywhere-Basic_.plist' test-i9-trimmed-2.test
	"test-i9-trimmed-2.test":
	I-9. Get-Printer-Attributes Operation to Serenity.localGet-Printer-Attributesattributes-charsetutf-8attributes-natural-languageenprinter-uriipp://Serenity.local:8991/ipp/print/_IPP-Everywhere-Basic_requesting-user-namesmittyrequested-attributesprinter-icons    Get-Printer-Attributes:
			attributes-charset (charset) = utf-8
			attributes-natural-language (naturalLanguage) = en
			printer-uri (uri) = ipp://Serenity.local:8991/ipp/print/_IPP-Everywhere-Basic_
			requesting-user-name (nameWithoutLanguage) = smitty
			requested-attributes (keyword) = printer-icons
		I-9. Get-Printer-Attributes Operation to Serenity.local              [successful-okattributes-charsetutf-8attributes-natural-languageenprinter-iconshttps://serenity.local.:8991/ipp/print/_IPP-Everywhere-Basic_/icon.pngFAIL]
			RECEIVED: 161 bytes in response
			status-code = successful-ok (successful-ok)
			attributes-charset (charset) = utf-8
			attributes-natural-language (naturalLanguage) = en
			printer-icons (uri) = https://serenity.local.:8991/ipp/print/_IPP-Everywhere-Basic_/icon.png
	EXPECTED: printer-icons WITH-ALL-VALUES "Serenity.local"GOT: printer-icons="https://serenity.local.:8991/ipp/print/_IPP-Everywhere-Basic_/icon.png"        EXPECTED: printer-icons WITH-ALL-VALUES "Serenity.local"
			GOT: printer-icons="https://serenity.local.:8991/ipp/print/_IPP-Everywhere-Basic_/icon.png"

Similar misbehavior is observed when no formatting arguments are provided and the "-P' argument is either provided or not. First observed with recent debug builds of the IPP Everywhere Self Certification test suite, where the output was corrupted and difficult to read.

**Comment by wifiprintguy** [comment]
The plot thickens - if you specify "-t -P file.plist" then the output is broken. If you specify "-P file.plist -t" then the CUPS test report format is unmolested.

**Comment by michaelrsweet** [comment]
Why specify both - the "-P" option already produces "-t" output in parallel.

**Comment by wifiprintguy** [comment]
Well currently the "-P" output doesn't seem to match the "-t" or "-vt" output, and has the extraneous stuff (i.e. instead of "[PASS]" or "[FAIL]" you get a bunch of stuff concatenated together that makes it very hard to read)

**Comment by michaelrsweet** [comment]
Hmm, confirmed with:

```
./ipptool -P t.plist ipps://HPC8D3FF16C2ED.local:443/ipp/print get-printer-attributes.test
"get-printer-attributes.test":
Get printer attributes using Get-Printer-AttributesGet-Printer-Attributesattributes-charsetutf-8attributes-natural-languageenprinter-uriipps://HPC8D3FF16C2ED.local:443/ipp/print    Get printer attributes using Get-Printer-Attributes                  [successful-okattributes-charsetutf-8attributes-natural-languageenprinter-uri-supportedipp://HPC8D3FF16C2ED.local/ipp/printipps://HPC8D3FF16C2ED.local:443/ipp/printuri-security-supportednonetlsuri-authentication-supportedrequesting-user-namerequesting-user-nameprinter-settable-attributes-supportedprinter-wifi-ssidprinter-wifi-passwordprinter-wifi-ssidSWEETprinter-wifi-stateprinter-nameHP16C2EDprinter-locationTammy's Officeprinter-more-infohttp://HPC8D3FF16C2ED.local./#hId-pgAirPrintprinter-infoHP ENVY 7640printer-dns-sd-nameHP ENVY 7640printer-make-and-modelHP ENVY 7640 seriesprinter-stateprinter-state-reasonsnoneprinter-state-messageipp-versions-supported1.01.12.0operations-supportedcharset-configuredus-asciicharset-supportedus-asciiutf-8natural-language-configuredengenerated-natural-language-supportedendocument-format-defaultapplication/octet-streamdocument-format-supportedapplication/vnd.hp-PCLimage/jpegapplication/PCLmimage/urfimage/pwg-rasterapplication/octet-streamdocument-format-version-supportedPCL3GUIPJLAutomaticJPEGPCLMAppleRasterPWGRasterprinter-is-accepting-jobsqueued-job-countpdl-override-supportedattemptedprinter-up-timeprinter-current-timecompression-supportednonedeflategzipcolor-supportedjob-creation-attributes-supportedcopiesfinishingssidesorientation-requestedmediaprint-qualityprinter-resolutionoutput-binmedia-coloutput-modeprint-content-optimizepclm-source-resolutionprint-color-modeipp-attribute-fidelityjob-namepage-rangesmultiple-document-handlingprint-rendering-intentprint-scalingreference-uri-schemes-supportedhttphttpsprinter-device-idMFG:HP;MDL:ENVY 7640 series;CMD:PCL3GUI,PJL,Automatic,JPEG,PCLM,AppleRaster,PWGRaster,DW-PCL,802.11,802.3,DESKJET,DYN;CLS:PRINTER;DES:E4W43A;CID:HPIJVIPAV4;LEDMDIS:USB#FF#CC#00,USB#07#01#02,USB#FF#04#01;IPP-HTTP:T;IPP-E:FF-CC-00,07-01-02,FF-04-01;SN:TH69M4X0Y1063T;S:038088C484400001002c240005ac140003c;Z:05000009000009,12000,17000000000000,180;printer-alertprinter-alert-descriptionsupplyInkLevelChangesupplyInkLevelChangegenuineHPinkSystemReadyinkSystemReadyreadyprocessingreadyprocessingreadyprocessingreadyprocessingreadyprocessingreadyprocessingreadyprocessingsupplyInkLevelChangereadyprocessingreadyinPowerSaveprinter-uuidurn:uuid:1c852a4d-b800-1f08-abcd-c8d3ff16c2edlandscape-orientation-requested-preferredjob-constraints-supportedresolver-nameduplex-sizessidestwo-sided-short-edgetwo-sided-long-edgemediana_legal_8.5x14inna_govt-letter_8x10inna_invoice_5.5x8.5iniso_a5_148x210mmjis_b5_182x257mmiso_a6_105x148mmiso_a6_105x148mmna_index-4x6_4x6inna_index-5x8_5x8inna_index-3x5_3x5inna_monarch_3.875x7.5inna_number-10_4.125x9.5iniso_dl_110x220mmiso_c5_162x229mmiso_c6_114x162mmna_a2_4.375x5.75injpn_chou3_120x235mmna_5x7_5x7inna_index-4x6_4x6inoe_photo_4x8inoe_photo_4x12inom_small-photo_100x150mmna_foolscap_8.5x13injob-resolvers-supportedresolver-nameduplex-sizessidesone-sidedipp-features-supportedairprint-1.4ipp-everywherewhich-jobs-supportedcompletednot-completedalljob-ids-supportedrequesting-user-uri-supportedmultiple-operation-time-out-actionprocess-jobprinter-geo-locationgeo:46.603889,-81.054722preferred-attributes-supportedprinter-config-change-date-timeprinter-config-change-timeprinter-state-change-date-timeprinter-state-change-timedocument-password-supportedprinter-kinddocumentenvelopephotopostcardmedia-supportedna_executive_7.25x10.5inna_letter_8.5x11inna_legal_8.5x14inna_govt-letter_8x10inna_invoice_5.5x8.5iniso_a5_148x210mmiso_a4_210x297mmjis_b5_182x257mmjpn_hagaki_100x148mmiso_a6_105x148mmna_index-4x6_4x6inna_index-5x8_5x8inna_index-3x5_3x5inna_monarch_3.875x7.5inna_number-10_4.125x9.5iniso_dl_110x220mmiso_c5_162x229mmiso_c6_114x162mmna_a2_4.375x5.75injpn_chou3_120x235mmna_5x7_5x7inoe_photo_4x8inoe_photo_4x12iniso_b5_176x250mmom_small-photo_100x150mmna_foolscap_8.5x13incustom_min_3x5incustom_max_8.5x14inmedia-defaultna_letter_8.5x11inmedia-col-supportedmedia-typemedia-sizemedia-top-marginmedia-left-marginmedia-right-marginmedia-bottom-marginmedia-sourcemedia-size-namemedia-col-defaultmedia-sizex-dimensiony-dimensionmedia-top-marginmedia-bottom-marginmedia-left-marginmedia-right-marginmedia-sourcemainmedia-typestationerymedia-left-margin-supportedmedia-right-margin-supportedmedia-top-margin-supportedmedia-bottom-margin-supportedmedia-source-supportedmainphotoautomedia-type-supportedstationeryphotographic-glossycom.hp.brochure-glossycom.hp.brochure-mattemedia-size-supportedx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionx-dimensiony-dimensionmedia-readyna_letter_8.5x11inna_index-4x6_4x6inmedia-col-readymedia-sizex-dimensiony-dimensionmedia-top-marginmedia-bottom-marginmedia-left-marginmedia-right-marginmedia-sourcemainmedia-typestationerymedia-sizex-dimensiony-dimensionmedia-top-marginmedia-bottom-marginmedia-left-marginmedia-right-marginmedia-sourcemainmedia-typestationerymedia-sizex-dimensiony-dimensionmedia-top-marginmedia-bottom-marginmedia-left-marginmedia-right-marginmedia-sourcemainmedia-typestationerymedia-sizex-dimensiony-dimensionmedia-top-marginmedia-bottom-marginmedia-left-marginmedia-right-marginmedia-sourcephotomedia-typephotographic-glossymedia-sizex-dimensiony-dimensionmedia-top-marginmedia-bottom-marginmedia-left-marginmedia-right-marginmedia-sourcephotomedia-typephotographic-glossypages-per-minutepages-per-minute-colorjpeg-k-octets-supportedjpeg-x-dimension-supportedjpeg-y-dimension-supportedpdf-versions-supportednoneurf-supportedCP1MT1-2-8-9-10-11PQ3-4-5RS300-600SRGB24OB9OFU0W8-16DEVW8-16DEVRGB24-48ADOBERGB24-48DM3IS1-7V1.4marker-namestri-color inkblack inkmarker-colors#00FFFF#FF00FF#FFFF00#000000marker-typesink-cartridgeink-cartridgemarker-low-levelsmarker-high-levelsmarker-levelsprinter-supplyprinter-supply-descriptionprinter-firmware-nameNSM2CN1537ARprinter-firmware-string-versionNSM2CN1537ARprinter-firmware-versionprinter-input-trayprinter-output-traycopies-defaultfinishings-defaultorientation-requested-supportedorientation-requested-defaultprint-quality-defaultprinter-resolution-defaultsides-defaultone-sidedoutput-bin-defaultface-upoutput-mode-defaultautoprint-color-mode-defaultautomultiple-document-handling-defaultseparate-documents-uncollated-copiesnumber-up-defaultpresentation-direction-number-up-defaulttoright-tobottomprint-rendering-intent-defaultautoprint-scaling-defaultautocopies-supportedfinishings-supportedprint-quality-supportedprinter-resolution-supportedsides-supportedone-sidedtwo-sided-short-edgetwo-sided-long-edgeoutput-bin-supportedface-upoutput-mode-supportedautoauto-monochromemonochromecolorprint-color-mode-supportedautoauto-monochromemonochromecolorprocess-monochromepage-ranges-supportedmultiple-document-handling-supportedseparate-documents-uncollated-copiesseparate-documents-collated-copiesnumber-up-supportedpresentation-direction-number-up-supportedtoright-tobottomprint-rendering-intent-supportedautoperceptualprint-scaling-supportedautoauto-fitfillfitnoneprinter-iconshttp://HPC8D3FF16C2ED.local/webApps/images/printer-small.pnghttp://HPC8D3FF16C2ED.local/webApps/images/printer.pnghttp://HPC8D3FF16C2ED.local/webApps/images/printer-large.pngprinter-supply-info-urihttp://HPC8D3FF16C2ED.local/SureSupplyMobile/print-content-optimize-defaultautoprint-content-optimize-supportedautophotographicstexttext-and-graphicspwg-raster-document-sheet-backrotatedpwg-raster-document-type-supportedsgray_8srgb_8adobe-rgb_8sgray_16srgb_16adobe-rgb_16rgb_8rgb_16pwg-raster-document-resolution-supportedepcl-version-supported1.0manual-duplex-supportedpclm-source-resolution-supportedpclm-source-resolution-defaultpclm-strip-height-supportedpclm-strip-height-preferredpclm-raster-back-siderotatedpclm-compression-method-preferredrlejpegflatedocument-format-varying-attributescopiesprinter-get-attributes-supporteddocument-formatprinter-organizationprinter-organizational-unitidentify-actions-defaultdisplayidentify-actions-supporteddisplaysoundpinlimit-operations-supportedmultiple-operation-time-outmultiple-document-jobs-supportedPASS]
```

**Comment by michaelrsweet** [comment]
The plist's key and value fields are empty - looks like they are not getting written to the output file.

**Comment by michaelrsweet** [comment]
OK, print_xml_string was sending its output to stdout instead of the output file; not sure when this bug got introduced, but fixed now...

[master a802a3b] Fix ipptool -P output (Issue #46)

**Comment by michaelrsweet** [comment]
(I'll mirror everything to the ippeveselfcert repo as soon as I resolve one other libcups bug...)
